common/src/main/java/org/opensearch/ml/common/MLModel.java

```diff
@@ -28,23 +34,71 @@ public class MLModel implements ToXContentObject {
     public static final String OLD_MODEL_CONTENT_FIELD = "content";
     public static final String MODEL_CONTENT_FIELD = "model_content";
 
+    public static final String DESCRIPTION_FIELD = "description";
+    public static final String MODEL_FORMAT_FIELD = "model_format";
+    public static final String MODEL_STATE_FIELD = "model_state";
+    public static final String MODEL_CONTENT_SIZE_IN_BYTES_FIELD = "model_content_size_in_bytes";
+    public static final String MODEL_CONTENT_HASH_FIELD = "model_content_hash";
+    public static final String MODEL_CONFIG_FIELD = "model_config";
+    public static final String CREATED_TIME_FIELD = "created_time";
+    public static final String LAST_UPLOADED_TIME_FIELD = "last_uploaded_time";
+    public static final String LAST_LOADED_TIME_FIELD = "last_loaded_time";
+    public static final String LAST_UNLOADED_TIME_FIELD = "last_unloaded_time";
+
+    public static final String MODEL_ID_FIELD = "model_id";
+    public static final String CHUNK_NUMBER_FIELD = "chunk_number";
+    public static final String TOTAL_CHUNKS_FIELD = "total_chunks";
+
     private String name;
     private FunctionName algorithm;
     private Integer version;
     private String content;
     private User user;
 
+    @Setter
+    private String description;
+    private MLModelFormat modelFormat;
+    private MLModelState modelState;
+    private Long modelContentSizeInBytes;
+    private String modelContentHash;
+    private MLModelConfig modelConfig;
+    private Instant createdTime;
+    private Instant lastUploadedTime;
+    private Instant lastLoadedTime;
+    private Instant lastUnloadedTime;
+
+    @Setter
+    private String modelId; // model chunk doc only
+    private Integer chunkNumber; // model chunk doc only
+    private Integer totalChunks; // model chunk doc only
+
     @Builder
-    public MLModel(String name, FunctionName algorithm, Integer version, String content, User user) {
+    public MLModel(String name, FunctionName algorithm, Integer version, String content, User user, String description, MLModelFormat modelFormat, MLModelState modelState, Long modelContentSizeInBytes, String modelContentHash, MLModelConfig modelConfig, Instant createdTime, Instant lastUploadedTime, Instant lastLoadedTime, Instant lastUnloadedTime, String modelId, Integer chunkNumber, Integer totalChunks) {
```

Comment:
`version` -> what if we want to have a version like 2.1?

Reply:
Good point. The `version` part was int from initial release. I plan to revise that part later to not change this backport PR. I will create a Github issue to track this.

Reply:
https://github.com/opensearch-project/ml-commons/issues/472